app/javascript/src/PaymentGateways/braintree/BraintreeForm.jsx

```diff
@@ -54,12 +56,20 @@ const BraintreeForm = ({
     }
   }, [braintreeNonceValue])
 
+  const get3DSecureError = (response) => {
+    return (
+      response.name === 'BraintreeError' ||
+        !!response.threeDSecureInfo.status.match(/authenticate_(attempt_)?successful/) === false
+        ? CC_ERROR_MESSAGE
+        : null
+    )
+  }
```

Comment:
If we consider that `response.status` could be `undefined` and thus treat it as an error, then it could be:

```suggestion
  const get3DSecureError = (response) => (
      response.name === 'BraintreeError' ||
        !response.status && !!response.threeDSecureInfo.status.match(/authenticate_(attempt_)?successful/) === false
        ? CC_ERROR_MESSAGE
        : null
    )

```

It's a pity we don't have types defined for the specific statuses, nor a list of them. Pattern matching is OK for now to identify them, but would be better to assert the specific statuses / errors in the future... If Braintree doesn't provide them, we should write them ourselves based on Braintree API documentation.

Reply:
I agree, but there are no types provided by Braintree, we need to craft them. That's out of scope of this quick-fix.